charmstore.go

```diff
@@ -359,11 +359,21 @@ func (s *CharmStore) GetResource(id *charm.URL, revision int, name string) (resu
 	}, nil
 }
 
-// Publish tells the charmstore to mark the given charm as published with the given resource revisions.
-func (s *CharmStore) Publish(id *charm.URL, resources map[string]int) (*charm.URL, error) {
+// Publish tells the charmstore to mark the given charm as published with the
+// given resource revisions to the given channels. If no channels are specified,
```

Comment:
I'm not entirely sure if we should do the "no channels==stable" thing at this level. It seems to me like that's more a policy decision for the outer level to take. This method could work OK as a very thin layer over the actual API endpoint, with no logic to speak of.